tools/targets/ARM_MUSCA_A1/ARM_MUSCA_A1_TFM.py

```diff
@@ -0,0 +1,103 @@
+ #
+ # Copyright (c) 2018 Arm Limited
+ #
+ # Licensed under the Apache License Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ #     http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing software
+ # distributed under the License is distributed on an "AS IS" BASIS
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+ #
+
+try:
+ import Crypto
+except ImportError, e:
+ print('Please install "pycryptodome" Python module! ("pip install --user pycryptodome")')
+ quit()
+
+import sys
+import os.path
+from os.path import(
+    join,
+    dirname
+)
+run_dir = dirname(os.path.realpath(__file__))
+sys.path.append(run_dir)
+
+import assemble
+import imgtool
+import re
+
+def musca_tfm_bin(t_self, mbed_ns_binary_file):
+    #Input files and paths
+    mcuboot_binary_file_name = "mcuboot.bin"
+    tfm_binary_file_name = "tfm_s.bin"
+    flash_layout_file_name = "flash_layout.h"
+    musca_tfm_rel_path = join("targets", "TARGET_ARM_SSG", "TARGET_MUSCA_A1", "TFM")
+
+    #Temporary files
+    mbed_ns_file_name, ext = os.path.splitext(os.path.basename(mbed_ns_binary_file))
+    concatenated_binary_file_name = mbed_ns_file_name + "_tfm.bin"
+    signed_binary_file_name = mbed_ns_file_name + "_signed.bin"
+
+    build_dir = dirname(mbed_ns_binary_file)
+    mbed_os_dir = dirname(dirname(dirname(run_dir)))
+    tfm_musca_dir = join(mbed_os_dir, musca_tfm_rel_path)
+    mcuboot_binary_file_name = join(tfm_musca_dir, mcuboot_binary_file_name)
+    tfm_binary_file = join(tfm_musca_dir, tfm_binary_file_name)
+    flash_layout_file = join(tfm_musca_dir, flash_layout_file_name)
+
+    concatenated_binary_file = join(build_dir, concatenated_binary_file_name)
+    signed_binary_file = join(build_dir, signed_binary_file_name)
+    temporary_files = [concatenated_binary_file, signed_binary_file]
+
+    #1. Run assemble.py to concatenate secure TFM and non-secure mbed binaries
+    sys.argv = ['assemble.py',
```

Comment:
This seems to be a pretty non-standard way of passing arguments to a function. Instead of setting `sys.argv`, would it be possible to provide these parameters to the relevant function in the `assemble` module?

Reply:
To add to this, `sys.argv` is destructively modified; the old value is lost. At the very least, don't do that.